test/modes_internal_test.c

```diff
@@ -197,76 +188,65 @@ static int execute_cts128(CTS128_FIXTURE fixture)
-
-    if (!TEST_mem_eq(iv, sizeof(iv), vector + len - tail, sizeof(iv)))
+    fixture->encrypt_block(test_input, ciphertext, len, encrypt_key_schedule,
+                           iv, (block128_f)AES_encrypt);
+    if (!TEST_mem_eq(ciphertext, len, vector, len)
+            || !TEST_mem_eq(iv, sizeof(iv), vector + len - tail, sizeof(iv)))
         return 0;
 
     /* test block-based decryption */
     memcpy(iv, test_iv, test_iv_len);
-    fixture.decrypt_block(ciphertext, cleartext, len,
-                          decrypt_key_schedule, iv,
-                          (block128_f)AES_decrypt);
-    if (!TEST_mem_eq(cleartext, len, test_input, len))
-        return 0;
-    if (!TEST_mem_eq(iv, sizeof(iv), vector + len - tail, sizeof(iv)))
+    fixture->decrypt_block(ciphertext, cleartext, len, decrypt_key_schedule,
+                           iv, (block128_f)AES_decrypt);
+    if (!TEST_mem_eq(cleartext, len, test_input, len)
+            || !TEST_mem_eq(iv, sizeof(iv), vector + len - tail, sizeof(iv)))
         return 0;
 
     /* test streamed encryption */
     memcpy(iv, test_iv, test_iv_len);
-    fixture.encrypt(test_input, ciphertext, len, encrypt_key_schedule,
-                    iv, (cbc128_f) AES_cbc_encrypt);
-    if (!TEST_mem_eq(ciphertext, len, vector, len))
-        return 0;
-    if (!TEST_mem_eq(iv, sizeof(iv), vector + len - tail, sizeof(iv)))
+    fixture->encrypt(test_input, ciphertext, len, encrypt_key_schedule,
+                     iv, (cbc128_f) AES_cbc_encrypt);
+    if (!TEST_mem_eq(ciphertext, len, vector, len)
+            || !TEST_mem_eq(iv, sizeof(iv), vector + len - tail, sizeof(iv)))
         return 0;
 
     /* test streamed decryption */
     memcpy(iv, test_iv, test_iv_len);
-    fixture.decrypt(ciphertext, cleartext, len, decrypt_key_schedule, iv,
-                    (cbc128_f)AES_cbc_encrypt);
-    if (!TEST_mem_eq(cleartext, len, test_input, len))
-        return 0;
-    if (!TEST_mem_eq(iv, sizeof(iv), vector + len - tail, sizeof(iv)))
+    fixture->decrypt(ciphertext, cleartext, len, decrypt_key_schedule, iv,
+                     (cbc128_f)AES_cbc_encrypt);
+    if (!TEST_mem_eq(cleartext, len, test_input, len)
+            || !TEST_mem_eq(iv, sizeof(iv), vector + len - tail, sizeof(iv)))
         return 0;
 
     return 1;
 }
 
 static int test_cts128(int idx)
 {
-    SETUP_TEST_FIXTURE(CTS128_FIXTURE, setup_cts128);
-    fixture.transform_output = transform_output;
-    fixture.encrypt_block = CRYPTO_cts128_encrypt_block;
-    fixture.encrypt = CRYPTO_cts128_encrypt;
-    fixture.decrypt_block = CRYPTO_cts128_decrypt_block;
-    fixture.decrypt = CRYPTO_cts128_decrypt;
-    fixture.case_name = "cts128";
-    fixture.num = idx;
-    EXECUTE_TEST_NO_TEARDOWN(execute_cts128);
+    static const CTS128_FIXTURE fixture_cts128 = {
+        "cts128", transform_output,
+        CRYPTO_cts128_encrypt_block, CRYPTO_cts128_encrypt,
+        CRYPTO_cts128_decrypt_block, CRYPTO_cts128_decrypt
+    };
```

Comment:
On side note, this is like worst parametrization ever. I mean it's more natural to collect parameters in one place, but here it's spread all over the code. For example, looking at this structure one can't tell that we exercise it with AES. What does transform_output do? Essentially code is virtually unparseable for unaware reader...